### PR TITLE
Retrieve user permission and roles from organization roles.

### PR DESCRIPTION
**Pull request by wenodpathirana**

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/dao/RoleManagementDAO.java [hunk 1]
```diff
@@ -69,6 +69,16 @@ List<Role> getOrganizationRoles(String organizationId, int count, List<Expressio
                                     List<String> operators, String cursor, String direction)
             throws OrganizationManagementServerException;
 
+    /**
+     *
+     * @param userId
+     * @param organizationId
+     * @return
+     * @throws OrganizationManagementServerException
+     */
```
[the remark this answers is not in the document]
Comment (wenodpathirana) [reply components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/dao/RoleManagementDAO.java]: fixed

File: components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/dao/RoleManagementDAOImpl.java [hunk 2]
```diff
@@ -285,6 +288,42 @@ public List<Role> getOrganizationRoles(String organizationId, int count, List<Ex
         }
     }
 
+    @Override
+    public List<Role> getUserOrganizationRoles(String userId, String organizationId)
+            throws OrganizationManagementServerException {
+        NamedJdbcTemplate namedJdbcTemplate = getNewTemplate();
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/dao/RoleManagementDAOImpl.java:294]: leave new line after method signature

File: components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/dao/RoleManagementDAOImpl.java [hunk 3]
```diff
@@ -285,6 +288,42 @@ public List<Role> getOrganizationRoles(String organizationId, int count, List<Ex
...
+        try {
+            return namedJdbcTemplate.executeQuery(GET_USER_ORGANIZATION_ROLES,
+                    (resultSet, rowNumber) -> new Role(resultSet.getString(DB_SCHEMA_COLUMN_NAME_UM_ROLE_ID),
+                            resultSet.getString(DB_SCHEMA_COLUMN_NAME_UM_ROLE_NAME)),
+                    namedPreparedStatement -> {
+                        namedPreparedStatement.setString(DB_SCHEMA_COLUMN_NAME_UM_ORG_ID, organizationId);
+                        namedPreparedStatement.setString(DB_SCHEMA_COLUMN_NAME_UM_USER_ID, userId);
+                    });
+        } catch (DataAccessException e) {
+            throw handleServerException(ERROR_CODE_ERROR_RETRIEVING_ORGANIZATION_PERMISSIONS, e, organizationId,
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/dao/RoleManagementDAOImpl.java:304]: Shouldn't this be some other error message ERROR_CODE_ERROR_RETRIEVING_ORGANIZATION_ROLES

Comment (wenodpathirana) [reply components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/dao/RoleManagementDAOImpl.java]: fixed

**Pull request merged**